libs/core/src/lib/action-sheet/action-sheet-item/action-sheet-item.component.ts

```diff
@@ -0,0 +1,117 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    EventEmitter,
+    HostListener,
+    Input,
+    Output,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { KeyboardSupportItemInterface } from '../../utils/interfaces/keyboard-support-item.interface';
+import { ButtonComponent } from '../../button/button.component';
+
+
+
+/**
+ * A component used to enforce a certain layout for the action sheet.
+ * ```html
+ * <fd-action-sheet>
+ *     <fd-action-sheet-control>Control Element</fd-action-sheet-control>
+ *     <fd-action-sheet-body>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *          <fd-action-sheet-item>Action Sheet Body</fd-action-sheet-item>
+ *     </fd-action-sheet-body>
+ * </fd-action-sheet>
+ * ```
+ */
+@Component({
+    // tslint:disable-next-line:component-selector
+    selector: '[fd-action-sheet-item]',
+    templateUrl: './action-sheet-item.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        'class': 'fd-action-sheet__item'
+    }
+})
+
+export class ActionSheetItemComponent implements KeyboardSupportItemInterface {
+
+    /** Sets text of button. */
+    @Input()
+    label: string;
+
+    /** Sets icon of action item. */
+    @Input()
+    glyph: string;
+
+    /** Indicate state of the button.*/
+    @Input()
+    negative = false;
+
+    /** Indicate if items should be in compact or compare mode. **/
+    @Input()
+    compact = false;
+
+    /** Display in mobile view. **/
+    @Input()
+    mobile = false;
+
+    /**Indicate if it's closing button **/
+    @Input()
+    isCloseButton = false;
+
+    /** @hidden */
+    @ViewChild(ButtonComponent)
+    buttonComponent: ButtonComponent;
+
+    /** @hidden **/
+    @Output()
+    keyDown: EventEmitter<KeyboardEvent> = new EventEmitter<KeyboardEvent>();
```

Comment:
No need for explicit typing when making a declaration and initialization at the same time.